telebot/__init__.py

```diff
@@ -5913,60 +5914,54 @@ def _run_middlewares_and_handler(self, message, handlers, middlewares, update_ty
                         else:
-                            if "data" in params:
-                                if len(params) == 2:
-                                    handler['function'](message, data)
-                                elif len(params) == 3:
-                                    handler['function'](message, data=data, bot=self)
-                                else:
-                                    logger.error("It is not allowed to pass data and values inside data to the handler. Check your handler: {}".format(handler['function']))
-                                    return
-                            
-                            else:
-
-                                data_copy = data.copy()
-                                
-                                for key in list(data_copy):
-                                    # remove data from data_copy if handler does not accept it
-                                    if key not in params:
-                                        del data_copy[key]
-                                if handler.get('pass_bot'): data_copy["bot"] = self
-                                if len(data_copy) > len(params) - 1: # remove the message parameter
-                                    logger.error("You are passing more data than the handler needs. Check your handler: {}".format(handler['function']))
-                                    return
-                                
-                                handler["function"](message, **data_copy)
-                        break
-        except Exception as e:
-            handler_error = e
-            if self.exception_handler:
-                self.exception_handler.handle(e)
-            else: logging.error(str(e))
-        
+                            logger.error("It is not allowed to pass data and values inside data to the handler. Check your handler: {}".format(handler['function']))
+                            return
+                    else:
+                        data_copy = data.copy()
+                        for key in list(data_copy):
+                            # remove data from data_copy if handler does not accept it
+                            if key not in params:
+                                del data_copy[key]
+                        if handler.get('pass_bot'):
+                            data_copy["bot"] = self
+                        if len(data_copy) > len(params) - 1: # remove the message parameter
+                            logger.error("You are passing more parameters than the handler needs. Check your handler: {}".format(handler['function']))
+                            return
+                        handler["function"](message, **data_copy)
+                    break
+            except Exception as e:
+                handler_error = e
+                if self.exception_handler:
+                    self.exception_handler.handle(e)
+                else: logging.error(str(e))
```

Comment:
instead of just showing the error this one could show full traceback using traceback.format_exc